src/app/reporting/Engine.cpp

```diff
@@ -47,6 +47,31 @@ void Engine::Shutdown()
     mGlobalDirtySet.ReleaseAll();
 }
 
+bool Engine::IsClusterDataVersionMatch(ClusterInfo * aDataVersionFilterList, const ConcreteReadAttributePath & aPath)
+{
+    bool existPathMatch       = false;
+    bool existVersionMismatch = false;
+    for (auto filter = aDataVersionFilterList; filter != nullptr; filter = filter->mpNext)
+    {
+        if (aPath.mEndpointId == filter->mEndpointId && aPath.mClusterId == filter->mClusterId)
+        {
+            existPathMatch = true;
+        }
+
+        if (aPath.mEndpointId == filter->mEndpointId && aPath.mClusterId == filter->mClusterId &&
+            !(filter->mDataVersion.HasValue() &&
+              IsClusterDataVersionEqual(filter->mEndpointId, filter->mClusterId, filter->mDataVersion.Value())))
+        {
+            existVersionMismatch = true;
```

Comment:
```suggestion

            if (!IsClusterDataVersionEqual(filter->mEndpointId, filter->mClusterId, filter->mDataVersion.Value()))
            {
                existVersionMismatch = true;
            }
```
Because if a ClusterInfo is in the data version filter list its mDataVersion.HasValue() has to be true.  Otherwise we would not have put it in the list.

Reply:
Agree, done, thanks